docs/design/interoperability/README.md

```diff
@@ -29,4 +50,209 @@ more detail.
+like `long` and `long long`, or `T*` and `T&`, which is critical for correct
+overload resolution and template instantiation.
+
+## C++ Interoperability Model: Introduction and Principles
+
+### The "Successor Language" Mandate
+
+The design of Carbon's C++ interoperability is governed by its foundational
+goal: [to be a successor language](/README.md), not merely a language with a
+foreign function interface (FFI). This mandate dictates a design that moves
+beyond the C-style FFI adopted by most modern languages and instead provides
+"seamless, bidirectional interoperability". The objective is to support deep
+integration with existing C++ code, encompassing its most complex features,
+"from inheritance to templates".
+
+This goal has profound implications for the Carbon compiler and language
+semantics. It requires that C++ is not treated as a "foreign" entity. Instead,
+Carbon's semantic model must be _co-designed_ to understand, map, and interact
+with C++'s semantic constructs—including templates, class hierarchies, and
+complex overload resolution—with high fidelity. The interoperability layer must,
+therefore, operate at the semantic analysis (SemIR) level, not just at the
+linking (ABI) level. This document specifies the design of this semantic
+contract.
+
+### The `Cpp` Associated Type: The Interop Trigger
+
+A core mechanism in this design is the `Cpp` associated type. This concept
+defines the "trigger" that activates C++-specific semantic rules within the
+Carbon compiler. Any operation involving a type that is designated as a `Cpp`
+associated type will invoke the specialized interoperability logic, such as the
+operator model detailed in
+[The Operator Interoperability Model Section](#todo-the-operator-interoperability-model).
+
+A type is considered a `Cpp` associated type if its definition involves an
+imported C++ type in any of the following ways:
+
+1.  The C++ type itself (for example, `Cpp.Widget`).
+2.  A pointer to a C++ type (for example, `Cpp.Widget*`).
+3.  A Carbon generic type parameterized with a C++ type (for example,
+    `MyCarbonVector(Cpp.Widget)`).
+4.  A Carbon struct or class containing a C++ type as a member (for example,
+    `MyCarbonStruct { x: Cpp.Widget }`).
+
+This "pervasive" model of C++-awareness is a fundamental design choice. The C++
+semantics are not confined to a specific `unsafe` or `extern "C++"` block; they
+"infect" any Carbon type that composes them. For example, when the Carbon
+compiler instantiates a _Carbon_ generic type like `MyCarbonVector(Cpp.Widget)`,
+its type system must be aware that the `Cpp.Widget` parameter carries
+C++-specific rules. This mandates that Carbon's own generic system, struct
+layout logic, and operator lookup must query the type system for the presence of
+a `Cpp` associated type. If present, the compiler must branch to a different,
+more complex logic path (for example, C++-aware overload resolution). This
```

Comment:
Just to note, this is an example of an implementation detail that doesn't seem to fit in with a language design. Also I'll note that calling it "more complex" seems like you're making a values judgment.